config/jupyterhub_config.py

```diff
@@ -33,13 +33,32 @@
 c.JupyterHub.cookie_secret_file = f"{os.environ['JUPYTERHUB_SECRETS_DIR']}/jupyterhub_cookie_secret"
 c.JupyterHub.db_url = f"sqlite:///{os.environ['JUPYTERHUB_SECRETS_DIR']}/jupyterhub.sqlite"
 
-c.JupyterHub.spawner_class = VirtualEnvSpawner
-
 # Create a group to indicate users with read/write access to MinIO
 c.JupyterHub.load_groups = {
-    VirtualEnvSpawner.RW_MINIO_GROUP: [],
+    CustomDockerSpawner.RW_MINIO_GROUP: [],
 }
 
+c.JupyterHub.spawner_class = CustomDockerSpawner
+
+c.DockerSpawner.hub_connect_url = f"http://{os.environ['SPARK_DRIVER_HOST']}:{os.environ['NOTEBOOK_PORT']}"
+# Set the Docker image to use for user containers
+c.DockerSpawner.image = os.environ['JUPYTERHUB_USER_IMAGE']
+
+c.DockerSpawner.cmd = ['echo', 'Starting JupiterHub Single User Server With DockerSpawner ...']
+
+# Container resource limits
+c.DockerSpawner.cpu_limit = 4
+c.DockerSpawner.mem_limit = '16G'
+
+# The network name that Docker containers will use to communicate
+network_name = os.environ.get('NETWORK_NAME')
+if network_name:
+    c.DockerSpawner.network_name = network_name
+c.DockerSpawner.use_internal_ip = True
+environment = os.environ.get('ENVIRONMENT', 'prod').lower()
+c.DockerSpawner.remove = environment != 'dev'
```

Comment:
Can you add a comment describing the purpose of this line?

Reply:
👍

Reply:
Under what conditions does the container get removed?

Reply:
when the container stops (error, user can also manually restart spawner).

Reply:
> user can also manually restart spawner

Here you're talking about the user image right? Not the user image spawner? 

How would the user restart the spawner?

Can you update the comment to explain when the image is removed?

Reply:
👍

Reply:
>Here you're talking about the user image right? Not the user image spawner?

yea the user image. Not jupyterhub server. 

> How would the user restart the spawner?
On the home page, users can stop and restart server. Similar to NERSC. 

<img width="1723" alt="Screenshot 2024-09-25 at 2 05 18 PM" src="https://github.com/user-attachments/assets/2d0c8217-228a-462a-8447-b360b43656eb">